homeassistant/components/discord/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for Discord integration."""
+from __future__ import annotations
+
+import logging
+
+from aiohttp.client_exceptions import ClientConnectorError
+import nextcord
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DiscordFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Discord."""
+
+    async def async_step_reauth(self, config: dict[str, str]) -> FlowResult:
+        """Handle a reauthorization flow request."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Confirm reauth dialog."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema({vol.Required(CONF_TOKEN): str}),
+                errors={},
+            )
+
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            token = user_input[CONF_TOKEN]
+            name = user_input.get(CONF_NAME, DEFAULT_NAME)
+
+            error, unique_id = await _async_try_connect(token)
+            entry = await self.async_set_unique_id(unique_id)
+            if entry and self.source == config_entries.SOURCE_REAUTH:
+                self.hass.config_entries.async_update_entry(entry, data=user_input)
+                await self.hass.config_entries.async_reload(entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
```

Comment:
When doing re-auth, we need to make sure the newly added token results in the same unique ID as the current config entry (to verify the same account is being re-authed).